src/rust/src/wallet_scanner.rs

```diff
@@ -13,19 +13,30 @@ use memuse::DynamicUsage;
 use zcash_note_encryption::{batch, BatchDomain, Domain, ShieldedOutput, ENC_CIPHERTEXT_SIZE};
 use zcash_primitives::{
     block::BlockHash,
-    consensus, constants,
-    sapling::{
-        note_encryption::{PreparedIncomingViewingKey, SaplingDomain},
-        SaplingIvk,
-    },
+    consensus,
+    sapling::note_encryption::SaplingDomain,
     transaction::{
         components::{sapling::GrothProofBytes, OutputDescription},
         Transaction, TxId,
     },
 };
 
+use crate::{
+    note_encryption::{
+        parse_and_prepare_sapling_ivk, try_sapling_note_decryption, DecryptedSaplingOutput,
+    },
+    params::{network, Network},
+};
+
 #[cxx::bridge]
-mod ffi {
+pub(crate) mod ffi {
+    #[namespace = "wallet"]
+    pub(crate) struct SaplingShieldedOutput {
+        cmu: [u8; 32],
+        ephemeral_key: [u8; 32],
+        enc_ciphertext: [u8; 580],
```

Comment:
No outgoing ciphertext?

Reply:
Not needed for this FFI. If we need it in future, we can amend the type (it's not part of the public interface of `zcashd`).

Reply:
Scratch that, we are adding it because we now also use `zcash_note_encryption` for wallet output recovery (solely for `z_viewtransaction`).